nailgun/entities.py

```diff
@@ -5546,14 +5566,62 @@ def __init__(self, server_config=None, **kwargs):
                 str_type='alphanumeric',
                 length=(2, 30),  # min length is 2 and max length is arbitrary
                 unique=True
-            )
+            ),
+            'location': entity_fields.OneToManyField(Location),
+            'organization': entity_fields.OneToManyField(Organization),
         }
         self._meta = {
             'api_path': 'api/v2/roles',
             'server_modes': ('sat', 'sam'),
         }
         super(Role, self).__init__(server_config, **kwargs)
 
+    def create_payload(self):
+        """Wrap submitted data within an extra dict.
+
+        For more information, see `Bugzilla #1151220
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1151220>`_.
+
+        """
+        return {u'role': super(Role, self).create_payload()}
```

Comment:
Could you please provide some details why this change is needed? We didn't have extra dict for creating/updating the role and both operations were working for us. Or extra dict is needed for support of `location` & `organization` fields introduced in this PR?

Reply:
Thats true ... the extra dict is required for orgs and locations.